src/python/cache.py

```diff
@@ -308,9 +452,35 @@ def reference(self, name):
         ref = self.exists(name, return_ref=True)
         if ref is None:
             raise RuntimeError("Data buffer (nor alias) {} does not exist".format(name))
+        if Cache.use_fsbuffer and free_alloc and name in self._refs.keys():
+            self.destroy(name, remove_fsbuffer=False)
         return ref
 
 
+    def clone(self, name):
```

Comment:
I don't think this is what we had in mind during the discussion. We wanted a `copy` method that would return a plain Numpy ndarray. Operators using the ndarrays could then use the acquired copies as they like and have them garbage collected once they are done.